frontend/lib/resolution/Resolver.cpp

```diff
@@ -552,16 +552,12 @@ Resolver::ReceiverScopesVec Resolver::methodReceiverScopes(bool recompute) {
   return savedReceiverScopes;
 }
 
-const CompositeType* Resolver::methodReceiverType() {
+const QualifiedType Resolver::methodReceiverType() {
```

Comment:
I was recently working on const-checking in dyno. The method receiver might be const, or not, and that information will be in the `QualifiedType`.

However, it is possible that this information is not necessary at this point in resolution. I do not remember if we will have test failures e.g. if I undo this change.

Reply:
This method returning a `QualifiedType` as opposed to `CompositeType` makes sense to me, though I am also not certain if the extra info is strictly needed at this point. My question is why a `QualifiedType` instead of a `QualifiedType*`?

Reply:
`QualifiedType` is relatively small and reasonable to pass/return by value. It effectively contains two pointers and an integer kind. We could have it return `QualifiedType*` and use `nullptr` in the case it's called on a non-method. But IMO it's better to return by value & that's the closest to returning `CompositeType*` (since `QualifiedType` is just effectively wrapping the type pointer).